planner/types.go

```diff
@@ -33,39 +63,39 @@ type ProcedureSpec interface {
 type ProcedureKind string
 
 type Edges struct {
-	Pred []PlanNode
-	Succ []PlanNode
+	predecessors []PlanNode
+	successors   []PlanNode
 }
 
 func (e *Edges) Predecessors() []PlanNode {
-	return e.Pred
+	return e.predecessors
 }
 
 func (e *Edges) Successors() []PlanNode {
-	return e.Succ
+	return e.successors
 }
 
-func (e *Edges) AddSuccessor(node PlanNode) {
-	e.Succ = append(e.Succ, node)
+func (e *Edges) AddSuccessors(nodes ...PlanNode) {
+	e.successors = append(e.successors, nodes...)
```

Comment:
I think maybe these methods should also add the back-edges... if a node is adding a successor then the successor also needs to add a predecessor.  Not sure what the right set of helper methods are, but I think it will become more apparent as we do more rewriting.